### ISSUE TITLE
Casino

### ISSUE THREAD

**Issue by bowTiesAreCool** [meow56/Kixley#5]
see the issue from the old Alpha_Stages repo: https://github.com/Kixley/Alpha_Stages/issues/10. We removed this when Alpha 1.3 broke; I think we should re-add the casino.

**Comment by bowTiesAreCool** [comment]
Not actually entirely sure that issue has any meaningful information.

**Comment by meow56** [comment]
Yeah, definitely. At this point, it's probably easier to just redo the casino from scratch.

**Comment by bowTiesAreCool** [comment]
Yeah, although we might be able to salvage some of [the old code in the drive](https://docs.google.com/document/d/1R_MTnUciYjTQFsSu7KpCqx0lUixyo0fZyZMnZh5CZOE/edit). Also, I think we should somehow involve Scumbags and Warlords.

**Comment by meow56** [comment]
That'd be fun to make. Much easier with the buttons, too.

**Comment by meow56** [comment]
Hey, if you need help, I think all my stuff is done.

**Comment by meow56** [comment]
Hmm. How should AI work for Scumbags & Warlords? Should it just be a simple "if you have something do it"? Or...

**Comment by meow56** [comment]
For that matter, how does betting work? How do point system.

**Comment by meow56** [comment]
And also, when trading, should it just be highest/lowest, or should they get a choice?

**Comment by cbp2170** [comment]
So, trading should give a choice, but into the rankings:

I made [a google doc for the probability calculations](https://docs.google.com/document/d/1YUgcfPsXc4m6bturE8vlLfqXXCD0c1TqyxwPcdChZjA/edit), but I'm having trouble with the double doubles. The math I did generated a greater than 100% change of getting one (which is obviously false), so I tried a Monte Carlo simulation where I generate 100,000 distinct shuffles and see how many have double doubles, but this was throwing weird reference errors. I think the errors are from how I copied your sorting program, as I'm not entirely sure how it works. Anyway, could you look over my math and/or my Monte Carlo simulation and try to fix whatever errors I made? I uploaded the Monte Carlo simulation as a txt because GitHub didn't seem to support me uploading .js files.
[Monte-Carlo-Double-Double-Prob.txt](https://github.com/meow56/Kixley/files/2217546/Monte-Carlo-Double-Double-Prob.txt)

**Comment by meow56** [comment]
Couldn't find the math error, but I got the program working.
[ddoubles.txt](https://github.com/meow56/Kixley/files/2217704/ddoubles.txt)

**Comment by cbp2170** [comment]
Wait, isn't it running 500000 times, not 50000 times? Also the reason you got the same number is probably because we have `results / iterations` at the end, so they are literally the same number. But, .954% (adjusted for 500000 vs. 50000) is waaaaayyy less than 3.43% (quad chance), and either way, 9.54% is also significantly lower than 42.25%, so you'd think I'd have alreаdy suspected a massive probability difference just from playing experience. This leads to a few options: a) we did something wrong with the simulator. b) my run of 4 math is off (although I think it seems realistic based on playing). c) I am crazy and didn't notice the massive probability difference. d) I (and everyone I've played 4-person games with) am really lucky and got a ton of extra double doubles. In my experience playing the game, I'd say out of the three, quads are by far the least common, then slightly fewer double doubles than runs of four, but both significantly higher. Maybe you should check my run of four math?

**Comment by cbp2170** [comment]
Also as a side note, it seems based on commit titles you got the slots in. Not enough time to test right now, so if they work, can you click the button in the checklist?

**Comment by meow56** [comment]
Sadly, the slots are not done. Wasn't sure how they would work.

You're right, it is running 500000 times.

I'll be honest, probability is not my strong suit...

Oh no i goofed. findscumbags wants it sorted, not shuffled! Jeez, I guess when you run 100000 iterations, even improbable things happen...

Okay, updated numbers: ddoubles are 13.6056%. I also used the quads to check my work, and they work out! ~3.4162%. Finally, runs are ~9.811%?

**Comment by cbp2170** [comment]
Can you give me the updated program so I can test runs more?

**Comment by meow56** [comment]
Sorry, edited.

**Comment by meow56** [comment]
Hey, when does S&W end? And how does the points affect betting and stuff?

**Comment by cbp2170** [comment]
Uhhhhhhhh, S&W ends when it ends normally, but I'd say like 4-6 rounds in a game for betting.

**Comment by meow56** [comment]
;-; Yeah, but what should getting first net you???

**Comment by meow56** [comment]
Alright, I'll just say...

Getting first: *2
Getting second: *1 (ie no change)
Getting third: *0.5
Getting fourth: *0
